lib/internal/test_runner/harness.js

```diff
@@ -151,7 +151,10 @@ function setup(root) {
   const rejectionHandler =
     createProcessEventHandler('unhandledRejection', root);
   const coverage = configureCoverage(root, globalOptions);
-  const exitHandler = () => {
+  const exitHandler = async () => {
+    if (root.subtests.length === 0) {
```

Comment:
It might be good to include a comment here. The check could also include `(root.hooks.before.length > 0 || root.hooks.after.length > 0)` to avoid the extra work of calling `run()` with nothing to do.